x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/eql_query_edit/eql_query_edit.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import type { DataViewBase } from '@kbn/es-query';
+import type { FieldConfig } from '../../../../shared_imports';
+import { UseField } from '../../../../shared_imports';
+import type {
+  EqlFieldsComboBoxOptions,
+  EqlOptions,
+  FieldsEqlOptions,
+} from '../../../../../common/search_strategy';
+import { queryRequiredValidatorFactory } from '../../validators/query_required_validator_factory';
+import { debounceAsync } from '../../validators/debounce_async';
+import { eqlQueryValidatorFactory } from '../../validators/eql_query_validator_factory';
+import { EqlQueryBar } from './eql_query_bar';
+import * as i18n from './translations';
+import type { FieldValueQueryBar } from '../query_bar';
+
+interface EqlQueryEditProps {
+  path: string;
+  eqlFieldsComboBoxOptions?: EqlFieldsComboBoxOptions;
+  eqlOptions?: EqlOptions;
+  showEqlSizeOption?: boolean;
+  showFilterBar?: boolean;
+  dataView: DataViewBase;
+  required?: boolean;
+  loading?: boolean;
+  disabled?: boolean;
+  onEqlOptionsChange?: (field: FieldsEqlOptions, newValue: string | undefined) => void;
+  onValidityChange?: (arg: boolean) => void;
+  onValiditingChange?: (arg: boolean) => void;
+}
+
+export function EqlQueryEdit({
+  path,
+  eqlFieldsComboBoxOptions,
+  eqlOptions,
+  showEqlSizeOption = false,
+  showFilterBar = false,
+  dataView,
+  required,
+  loading,
+  disabled,
+  onEqlOptionsChange,
+  onValidityChange,
+  onValiditingChange,
```

Comment:
Looks like there's a typo. Did you mean `onIsValidatingChange`?

Reply:
Thanks for noticing. I copies this prop name from the existing `EqlQueryBar` component but it definitely makes sense to fix this typo.